src/Container/AgFeatureGrid/AgFeatureGrid.jsx

```diff
@@ -0,0 +1,1019 @@
+        if (node) {
+          node.setSelected(false);
+        }
+      } else {
+        selectedFeature.setStyle(selectStyle);
+
+        const node = this.getRowFromFeatureId(selectedFeature.ol_uid);
+        if (node) {
+          node.setSelected(true);
+        }
+      }
+    });
+  }
+
+  /**
+   * Removes the vector layer from the given map (if any).
+   */
+  deinitVectorLayer = () => {
+    const {
+      map
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    map.removeLayer(this._layer);
+  }
+
+  /**
+   * Unbinds the pointermove and click event handlers from the map (if given).
+   */
+  deinitMapEventHandlers = () => {
+    const {
+      map,
+      selectable
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    map.un('pointermove', this.onMapPointerMove);
+
+    if (selectable) {
+      map.un('singleclick', this.onMapSingleClick);
+    }
+  }
+
+  /**
+   * Returns the column definitions out of the attributes of the first
+   * given feature.
+   *
+   * @return {Array} The column definitions.
+   */
+  getColumnDefs = () => {
+    const {
+      attributeBlacklist,
+      features,
+      columnDefs,
+      selectable
+    } = this.props;
+
+    let columns = [];
+    const feature = features[0];
+
+    if (!(feature instanceof OlFeature)) {
+      return columns;
+    }
+
+    const props = feature.getProperties();
+
+    if (selectable) {
+      columns.push({
+        headerName: '',
+        checkboxSelection: true,
+        headerCheckboxSelection: true,
+        width: 28,
+        pinned: 'left',
+        lockPosition: true,
+        suppressMenu: true,
+        suppressSorting: true,
+        suppressFilter: true,
+        suppressResize: true,
+        suppressMovable: true
+      });
+    }
+
+    let index = -1;
+
+    Object.keys(props).forEach(key => {
+      if (attributeBlacklist.includes(key)) {
+        return;
+      }
+
+      if (props[key] instanceof OlGeomGeometry) {
+        return;
+      }
+
+      if (columnDefs[key].index !== undefined) {
+        index = columnDefs[key].index;
+      } else {
+        ++index;
+      }
+      columns[index] = {
+        headerName: key,
+        field: key,
+        ...columnDefs[key]
+      };
+    });
+
+    return columns;
+  }
+
+  /**
+   * Returns the table row data from all of the given features.
+   *
+   * @return {Array} The table data.
+   */
+  getRowData = () => {
+    const {
+      features
+    } = this.props;
+
+    let data = [];
+
+    features.forEach(feature => {
+      const properties = feature.getProperties();
+      const filtered = Object.keys(properties)
+        .filter(key => !(properties[key] instanceof OlGeomGeometry))
+        .reduce((obj, key) => {
+          obj[key] = properties[key];
+          return obj;
+        }, {});
+
+      data.push({
+        key: feature.ol_uid, //this.props.keyFunction(feature),
```

Comment:
Stray comment?

Reply:
Currently I'm not even sure, why this has been changed as I think it is still needed. So I would suggest to readd the call to `keyFunction`.

Reply:
Done.